src/main/java/org/wildfly/security/_private/ElytronMessages.java

```diff
@@ -215,4 +219,26 @@
 
     @Message(id = 56, value = "Invalid hex character")
     DecodeException invalidHexCharacter();
+
+    @LogMessage(level = Logger.Level.DEBUG)
+    @Message(id = 57, value = "JAAS authentication failed for principal %s")
+    void debugJAASAuthenticationFailure(Principal principal, @Cause Throwable cause);
+
+    @Message(id = 58, value = "Invalid principal type (expected %s, got %s)")
+    IllegalArgumentException invalidPrincipalType(Class<?> expectedType, Class<?> actualType);
+
+    @Message(id = 59, value = "Failed to create login context")
+    RealmUnavailableException failedToCreateLoginContext(@Cause Throwable cause);
+
+    @Message(id = 60, value = "Failed to instantiate custom CallbackHandler")
+    RealmUnavailableException failedToInstantiateCustomHandler(@Cause Throwable cause);
+
+    @Message(id = 61, value = "The Callback array cannot be null")
+    IllegalArgumentException invalidNullCallbackArray();
+
+    @Message(id = 62, value = "Credential cannot be converted to a password")
+    UnsupportedCallbackException failedToConvertCredentialToPassword(@Param Callback callback);
+
+    @Message(id = 63, value = "Unsupported Callback")
+    UnsupportedCallbackException unsupportedCallback(@Param Callback callback);
```

Comment:
Probably either FastUnsupportedCallbackException should be here, else ``CallbackUtil.unsupported(callback)`` should be used instead; many mechanisms will (for example) try a CredentialCallback then fall back to a PasswordCallback.